### DISCUSSION TITLE
Throw window to another Desktop (Not Screen)

### DISCUSSION THREAD

**Discussion by user-flynn7** [rxhanson/RectanglePro-Community#303] [Q&A]
Hi, 
I love Rectangle pro as its useful 99% of the times, The other 1% is that when I open a couple of programs, I feel the need to send a couple of non essential ones to a new desktop as shown below

![image](https://github.com/rxhanson/RectanglePro-Community/assets/54879113/3a7b8203-48fa-43ab-ae82-c8d6fdc1194d)

However I could not find any way to do this with rectangle pro, Is there any plan to support this? I am sure a lot of users will appreciate this feature

**Comment by rxhanson** [comment]
Thanks for the feedback. There is no support for moving windows to other Spaces/Desktops at this time.

https://github.com/rxhanson/Rectangle/discussions/402

**Comment by rxhanson** [comment]
In the most recent release, 3.0.12, I added in the move to next/prev space keyboard shortcuts.

**Comment by t4k** [comment]
This seems to be working great for most apps. I've tried a number of different shortcut combinations, but I cannot get any VS Code windows to move to the next space using my shortcuts. Any ideas?

**Reply by rxhanson** [reply]
Are you on v3.0.20 or higher? I updated it for VSCode in v3.0.20

**Reply by t4k** [reply]
I was on the 3.0.19 version. Updated and now it works. Thanks!

**Comment by user-flynn7** [comment]
hey @rxhanson. Thanks for bringing this much awaited feature. I had to move away from a mac so couldnt test it out. Although I am back on a mac now and for some reason the move to next space does not work for me. Please check out the video below. My cursor just goes to the top left of the screen and comes back to the original position.


https://github.com/rxhanson/RectanglePro-Community/assets/54879113/8af5c0d6-90c8-4f1b-aff9-f1b46ff26fd3

**Reply by allisonlindner** [reply]
@MrWick7796 if you changed the shortcut for move to next and previous space, that is the problem, change the shortcut to default and it will work.

**Reply by rxhanson** [reply]
Thanks for replying where I didn't! Indeed this would be the scenario, as Rectangle Pro is executing the default macOS keyboard shortcut for switching spaces. I'm not aware of a way to figure this out without user input as a 3rd party, but if there is a way to do that, I'll see if I can fix this.

**Reply by cyber-nic** [reply]
A "Next Space" / "Previous Space" tooltip would be handy. Thanks

**Reply by rxhanson** [reply]
@cyber-nic Good point, I'll add that in. 

If you do change the default macOS keyboard shortcuts for switching Spaces, you can adjust Rectangle Pro to use those with Terminal commands:

```
defaults write com.knollsoft.Hookshot nextSpaceShortcut -string "{\"keyCode\":<KEYCODE>,\"modifierFlags\":<MODIFIER_FLAGS>}"
defaults write com.knollsoft.Hookshot previousSpaceShortcut -string "{\"keyCode\":<KEYCODE>,\"modifierFlags\":<MODIFIER_FLAGS>}"
```
Replace `<KEYCODE>` and `<MODIFIER_FLAGS>` with the actual values. You can use an app like [Key Codes](https://apps.apple.com/us/app/key-codes/id414568915?mt=12) to figure these out.

Then restart the app afterwards.

My plan is to adjust the keyboard shortcut recorder to allow this to be input as a modifier keypress followed by the key (as opposed to executing the shortcut to record it). When I have that change in place, then this can be configured in the UI instead of via Terminal.

**Reply by MauScheff** [reply]
That just doesn't work... it's frustrating... I tried Key Codes, ChatGPT, and even reverse-engineering the shortcut from Rectangles Pro itself, have something like this: `defaults read com.knollsoft.Hookshot`:

```
...
   nextSpace =     {
        keyCode = 124;
        modifierFlags = 1572864;
    };
    nextSpaceShortcut = "{\\"keyCode\\":124,\\"modifierFlags\\":1310720}";
...
```

I restart the App, and the mouse does jump to the edge of the screen, but doesn't throw the window.

I use `Ctrl` for something else, so I can't rely on OSX default shortcuts. How can I gain access to this feature? It's one of the most important reasons I use Rectangles Pro...

**Reply by rxhanson** [reply]
Sorry, it's not my intent to make it frustrating. There is a missing piece of information in my comment here, where you need to add the modifier value for "fn" if the arrow key is part of the shortcut.  See this comment: https://github.com/rxhanson/RectanglePro-Community/issues/799#issuecomment-3536114454

My plan is still to update the shortcut recorder so that this can be recorded in the UI instead of using Terminal commands.

**Reply by kcrwfrd** [reply]
@rxhanson any chance of shortcuts to send a window to space by number?

It's such a game-changing feature in Yabai...unfortunately can't disable system integrity protection to use it on a work computer.

**Reply by rxhanson** [reply]
Unfortunately, that functionality would require using a private framework; something that I don't have plans to do in Rectangle Pro. I do have plans to build my own version of Spaces in Rectangle Pro, but it's still going to be a while before that makes it out the door.

**Reply by MauScheff** [reply]
> Sorry, it's not my intent to make it frustrating. There is a missing piece of information in my comment here, where you need to add the modifier value for "fn" if the arrow key is part of the shortcut. See this comment: [#799 (comment)](https://github.com/rxhanson/RectanglePro-Community/issues/799#issuecomment-3536114454)
> 
> My plan is still to update the shortcut recorder so that this can be recorded in the UI instead of using Terminal commands.

That worked! Thanks a lot!
